consensus/src/lib.rs

```diff
@@ -1,27 +1,624 @@
+            }
+            self.connect_one_block(&block_index)?;
+            return self.set_best_block_params(block_index);
+        }
+
+        if let Some(best_block_id) = best_block_id {
+            // Chain trust is higher than the best block
+            if self.highest_block_trust < block_index.chain_trust {
+                // is the main net?
+                if block_index.get_prev_block_id() == &Some(best_block_id) {
+                    // .. just connect the new block
+                    self.connect_one_block(&block_index)?;
+                    return self.set_best_block_params(block_index);
+                }
+                // It's not in the main net? Ok, let's do reorg
+                self.connect_blocks(&block_index)?;
+                return self.set_best_block_params(block_index);
+            }
+
+            // Equal chain trust or less
+            if self.highest_block_trust >= block_index.chain_trust {
+                self.store_index_without_connect(&block_index)?;
+            }
+        }
+        Ok(None)
+    }
+
+    fn get_block_proof(&self, _block: &Block) -> u64 {
+        //TODO: We have to make correct one
+        10
+    }
+
+    fn add_block_index(&mut self, block: &Block) -> Result<BlockIndex, BlockError> {
+        // TODO: Check for duplicate block index
+        let prev_block_index = self
+            .db_tx
+            .get_block_index(&block.get_prev_block_id().ok_or(BlockError::Unknown)?.into())
+            .map_err(BlockError::from)?;
+        // Set the block height
+        let height = match &prev_block_index {
+            Some(prev_block_index) => prev_block_index.height + 1,
+            None => BlockHeight::zero(),
+        };
+        // Set Time Max
+        let time_max = match &prev_block_index {
+            Some(prev_block_index) => {
+                std::cmp::max(prev_block_index.time_max, block.get_block_time())
+            }
+            None => block.get_block_time(),
+        };
+        // Set Chain Trust
+        let chain_trust = match &prev_block_index {
+            Some(prev_block_index) => prev_block_index.chain_trust,
+            None => 0,
+        } + self.get_block_proof(block);
+
+        let block_index = BlockIndex::new(block, chain_trust, height, time_max);
+        self.db_tx.set_block_index(&block_index).map_err(BlockError::from)?;
+        Ok(block_index)
+    }
+
+    #[allow(dead_code)]
+    fn accept_block(&mut self, block: &Block) -> Result<BlockIndex, BlockError> {
+        let block_index = self.add_block_index(block)?;
+        self.db_tx.add_block(block).map_err(BlockError::from)?;
+        self.check_block_index(&block_index)?;
+        Ok(block_index)
+    }
+
+    fn exist_block(&self, id: &Id<Block>) -> Result<bool, BlockError> {
+        Ok(self.db_tx.get_block_index(id)?.is_some())
+    }
+
+    fn check_block_index(&self, blk_index: &BlockIndex) -> Result<(), BlockError> {
```

Comment:
Btw there's inconsistent naming. Sometimes the block index is called `block_index` and here it's called `blk_index`